app/src/main/java/org/fossasia/openevent/app/module/main/MainActivity.java

```diff
@@ -207,7 +220,28 @@ private void loadFragment(int navItemId) {
                 fragment = EventDashboardFragment.newInstance(eventId);
                 break;
         }
-        fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).commit();
+
+        if(!isAtHome){
+            fragmentManager.popBackStack("root_fragment",0);
+            isAtHome=true;
+        }
+        if(atHome){
+            root_fragment_id = lastSelectedNavItemId;
+            fragment_tag = "root_fragment";
+            fragmentManager.beginTransaction().replace(R.id.fragment_container, fragment).addToBackStack(fragment_tag).commit();
+            atHome=false;
+        }
+        else {
+            isAtHome = false;
```

Comment:
Don't understand this at all. When `atHome` is false, both the first block will run and this will run as well. Hence why not merge them. Also, one makes `atHome = true` so the `if (atHome)` is ran, then why not change the condition or just add an OR statement? The whole logic is just weird

Reply:
First 'atHome=false' is to indicate the app ran for the first time and so this block won't run after first time.Now we have 1 item in backstack. Now when we select an item from navbar it will go to else case in which 'isAtHome' indicates that the pointer is not at homefragment rather on the other fragment thus making it false.